pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -1809,6 +1809,35 @@ public class Foo {
         </example>
     </rule>
 
+
+    <rule name="UseDiamondOperator"
+          language="java"
+          since="6.11.0"
+          message="Explicit type arguments can be replaced by Diamond Operator "
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_codestyle.html#usediamondoperator"
+          minimumLanguageVersion="1.7">
+        <description>Use the diamond operator to let the type be inferred</description>
+        <priority>1</priority>
+        <properties>
+            <property name="xpath">
+                <value>
+                    <![CDATA[
+//VariableInitializer
+//PrimaryExpression[not(PrimarySuffix)]
+[not(ancestor::ArgumentList)]
+/PrimaryPrefix/AllocationExpression[ClassOrInterfaceType[@AnonymousClass='false']/TypeArguments//ReferenceType[not(.//TypeArguments)]]
+|
+//StatementExpression[AssignmentOperator][PrimaryExpression/PrimaryPrefix[not(Expression)]]
+//PrimaryExpression[not(PrimarySuffix)]
+[not(ancestor::ArgumentList)]
+/PrimaryPrefix/AllocationExpression[ClassOrInterfaceType[@AnonymousClass='false']/TypeArguments//ReferenceType[not(.//TypeArguments)]]
+]]>
+                </value>
+            </property>
+        </properties>
+    </rule>
```

Comment:
We should add a short code example to make it clear, what exactly the rule will detect, and how it can be resolved:

```xml
    <example>
<![CDATA[
List<String> strings = new ArrayList<String>(); // unnecessary duplication of type parameters
List<String> stringsWithDiamond = new ArrayList<>(); // using the diamond operator is more concise
]]>
    </example>
```